### PR TITLE
Feature/java grep stream

### PR DESCRIPTION
**Pull request by vassairm**

### CODE REVIEW

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+   * @return files under the rootDir
+   */
+  @Override
+  public List<File> listFiles(String rootDir) {
+    List<File> fileList = new ArrayList<>();
+    File root = new File(rootDir);
+    File[] filesToCheck;
+    if (root.isDirectory()) {
+      filesToCheck = root.listFiles();
+      if (filesToCheck != null) {
```
Comment (yangdav9) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java:59]: Some redundancies - can do for(filesTocheck : root.listFiles) <-- if root.listFiles is NULL, should not execute the loop

Comment (vassairm) [reply core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java:59]: My understanding is that if root.listFiles returns null the foreach will hit a null pointer and die.
From the Java docs on File.listFiles:
`The array will be empty if the directory is empty. Returns null if this abstract pathname does not denote a directory, or if an I/O error occurs.`

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+   * Overrides JavaGrepImp.readLines(). This implementation uses java.nio.Files.
+   * @param file a File object which lines must be read from. This must be a file, not a directory
+   * @return a List of Strings containing all readable lines of the supplied file
+   */
+  @Override
+  public List<String> readLines(File file) {
+    List<String> lines = new ArrayList<>();
+    try {
+      lines = Files.readAllLines(file.toPath()); // Ed may not consider this "using streams"
+    } catch (IOException ex) {
```
Comment (andrew-laniak-fuoco) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java]: You are right, this is not.using streams.
Try again by creating a stream of just the one file.  Look up how to use tbe flatmap function to turn the file into a stream of strings (lines in the file).

**Review by andrew-laniak-fuoco — CHANGES_REQUESTED** [review]

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java (hunk 2 above)
Comment (vassairm) [reply core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java]: changed to `Files.lines(file.toPath()).collect(Collectors.toList());`

**Review by andrew-laniak-fuoco — APPROVED** [review]

**Pull request merged**